scanner/services/matcher.go

```diff
@@ -164,3 +170,44 @@ func (s *matcherService) notes(ctx context.Context, vr *v4.VulnerabilityReport)
 
 	return nil
 }
+
+func (s *matcherService) GetSBOM(ctx context.Context, req *v4.GetSBOMRequest) (*v4.GetSBOMResponse, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/service/matcher.GetSBOM")
+
+	if err := validators.ValidateGetSBOMRequest(req); err != nil {
+		return nil, errox.InvalidArgs.CausedBy(err)
+	}
+
+	// We only support container image resources for now.
+	digestStr, found := strings.CutPrefix(req.GetHashId(), "/v4/containerimage/")
+	if !found {
+		return nil, errox.InvalidArgs.CausedByf("invalid hash id %q, only container images are supported", req.GetHashId())
+	}
```

Comment:
can we just put this in the validator?

Reply:
That is where I wanted to put it, but a validator that also returns multiple fields that are used later didn't make sense, and wIthout the returns we'd be  'double validating' which is wastefule.

Good news is based on the outcomes from the bug bash I'll be removing this field entirely so this will go away.